contributing: add sections and update phrasing (#1365)
fluent/fluent-bit-docs: CONTRIBUTING.md

# Contributing to Fluent Bit docs

First of all, thanks for taking the time to read this guide. The fact that you're
here means you're interested in contributing to Fluent Bit, and we highly appreciate
your time.

This repository contains the files for the
[Fluent Bit documentation library](https://docs.fluentbit.io/). Keeping these docs
separate from the [main Fluent Bit repository](https://github.com/fluent/fluent-bit)
helps reduce the number of commits to the Fluent Bit source code and makes it
easier to maintain both projects.

Fluent Bit has a group of dedicated maintainers who oversee this repository,
including several technical writers. These writers will review any pull requests

> you open, so don't be afraid to contribute—even if you're not a writer by trade,
> your suggestions are valuable, and we'll help you wrangle any stray commas.

Reviewer: ```suggestion
you open, so don't be afraid to contribute—even if you're not a writer by trade.
Your suggestions are valuable, and we'll help you wrangle any stray commas.
```

little less run-on this way :D

Author: good suggestion, I like yours better 👍

[GitBook](https://docs.gitbook.com/). Unfortunately, GitBook doesn't support
local previews for contributors, but a Fluent Bit maintainer with a dedicated GitBook
account can verify that things are formatted correctly after you open a new pull
request.

Each `.md` file in this repository is a single page. You can use
[standard Markdown syntax](https://docs.gitbook.com/content-editor/editing-content/markdown)
to edit existing pages, or create a new `.md` file to add an additional page to
the docs library. If you create a new page, you'll also need to update
[GitBook's `SUMMARY.md` file](https://docs.gitbook.com/integrations/git-sync/content-configuration#structure)
(or ask a maintainer to update it for you).

## Workflow

After you open a pull request in this repo, a Fluent Bit maintainer will review
it, triage it, add comments or suggestions as needed, and then merge it. After
your changes are successfully merged into `master`, the docs site will update
within a few minutes.

### Stale pull requests

If you open a pull request that requires ongoing discussion or review, the
Fluent Bit maintainers will add a [`waiting-for-user` tag](#tags) to your pull

> request. This tag means that we're blocked from moving forward until you reply;
> to keep contributions from going stale, we'll wait [three weeks?] for your response,

Reviewer: ```suggestion
request. This tag means that we're blocked from moving forward until you reply.
To keep contributions from going stale, we'll wait [three weeks?] for your response,
```

You'll want to get that three weeks more concrete though, we don't want to merge it with a guess.

Author: yep, good call. per our Slack discussion I'm gonna bump this to six weeks (but in a separate commit so I can modify a few more lines at once)

> request. This tag means that we're blocked from moving forward until you reply;
> to keep contributions from going stale, we'll wait [three weeks?] for your response,
> and close the pull request if we don't hear back from you.

Author: ```suggestion
request. This tag means that we're blocked from moving forward until you reply;
to keep contributions from going stale, we'll wait 45 days for your response,
but we may close the pull request if we don't hear back from you by then.
```

practice in the Fluent Bit source code.

### Commit subjects

Use descriptive commit subjects that describe which file or interface you're
modifying.

For example, if you're modifying the Syslog output plugin doc, whose file is
located at [pipeline/outputs/syslog.md](https://github.com/fluent/fluent-bit-docs/blob/master/pipeline/outputs/syslog.md), this would be a descriptive commit subject:

`pipeline: outputs: syslog: fix grammar in examples`

Since this commit is prefixed with the relevant file path, it helps our maintainers
understand and prioritize your contribution.

### Set your email in Git

Make sure your email address is configured in your local Git environment. This
should be the same email address associated with your GitHub account.

For more information, refer to GitHub's guide to
[setting your commit email address in Git](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address#setting-your-commit-email-address-in-git).

### Sign off your commits

You must sign off your commits to certify your identity as the commit author. If

> you don't sign off your commits, our CI system will flat the pull request with a

Reviewer: ```suggestion
you don't sign off your commits, our CI system will flag the pull request with a
```
